nucliadb_models/nucliadb_models/search.py

```diff
@@ -776,6 +817,11 @@ class ChatRequest(BaseModel):
     security: Optional[
         RequestSecurity
     ] = SearchParamDefaults.security.to_pydantic_field()
+    rag: Optional[RAGOptions] = Field(
```

Comment:
This config feels a little complex. Is there a way to do it with less custom validation and less required None checks?

Maybe something like this?

```
class RagStrategy(pydantic.BaseModel):
    name: str

class SimpleStrategy(RagStrategy)
    name: str = Field("simple", const=True)

class ExtendeFieldStrategy(RagStrategy)
    name: str = Field("extended_fields", const=True)
    fields: list[str]

class ChatRequest(BaseModel):
    ...
    rag_strategies: list[RagStrategy] = [SimpleStrategy] # might need union here
```

Reply:
hm, I like it more! I will try, but IIRC fastapi doesn't know how to properly choose which model of the union has to deserialize to

Reply:
nice! I made it work, thanks for the tip!